src/test/java/com/salesforce/mirus/KafkaMonitorTest.java

```diff
@@ -138,4 +140,115 @@ public void shouldNotReBalanceIfOnlySourcePartitionOrderHasChanged() {
             new PartitionInfo("topic1", 0, null, null, null)));
     assertThat(kafkaMonitor.partitionsChanged(), is(false));
   }
+
+  @Test
+  public void shouldApplyTopicRenameTransforms() {
+    Map<String, String> properties = new HashMap<>();
+    properties.put(SourceConfigDefinition.TOPICS_REGEX.getKey(), "reroute.*");
+    properties.put(TaskConfigDefinition.CONSUMER_CLIENT_ID, "testId");
+    properties.put("transforms", "reroute");
+    properties.put("transforms.reroute.type", "org.apache.kafka.connect.transforms.RegexRouter");
+    properties.put("transforms.reroute.regex", "^reroute\\.outgoing$");
+    properties.put("transforms.reroute.replacement", "reroute.incoming");
+    properties.put("name", "mockKafkaMonitor");
+    properties.put("connector.class", "com.salesforce.mirus.MirusSourceConnector");
+    SourceConfig config = new SourceConfig(properties);
+
+    MockConsumer<byte[], byte[]> mockSource = new MockConsumer<>(OffsetResetStrategy.EARLIEST);
+    updateMockPartitions(mockSource, "reroute.outgoing", 1);
+
+    MockConsumer<byte[], byte[]> mockDest = new MockConsumer<>(OffsetResetStrategy.EARLIEST);
+    updateMockPartitions(mockDest, "reroute.incoming", 1);
+
+    TaskConfigBuilder taskConfigBuilder =
+        new TaskConfigBuilder(new RoundRobinTaskAssignor(), config);
+    KafkaMonitor monitor =
+        new KafkaMonitor(
+            mock(ConnectorContext.class), config, mockSource, mockDest, taskConfigBuilder);
+
+    monitor.partitionsChanged();
+    List<Map<String, String>> result = monitor.taskConfigs(3);
+
+    List<List<TopicPartition>> partitions =
+        result
+            .stream()
+            .map(i -> new TaskConfig(i).getInternalTaskPartitions())
+            .map(TopicPartitionSerDe::parseTopicPartitionList)
+            .collect(Collectors.toList());
+    assertThat(partitions.size(), is(1));
```

Comment:
If you're only doing a size-check on the collection, what is the value of running it through the transformation?

Reply:
If the transformation isn't properly applied, after `monitor.partitionsChanged()` is called above, no partitions will be mirrored and the collection will be empty. Based off the existing test checking that missing partitions aren't mirroed.

Reply:
I guess you're relying on the destination partition being pruned out out of the result set if the expected names don't match. Would be nice if there was a way to express that more directly in the test - is hard to follow what's going on here. 

Not sure, but would it help to externalize the `MissingPartitionsJmxReporter`class? That way we we can directly test whether partitions were considered missing?